NAM/activations.cpp

```diff
@@ -0,0 +1,35 @@
+#include "activations.h"
+
+std::unordered_map<std::string, activations::Activation*> activations::Activation::_activations = {
+  {"Tanh", new activations::ActivationTanh()},
+  {"Hardtanh", new activations::ActivationHardTanh()},
+  {"Fasttanh", new activations::ActivationFastTanh()},
+  {"ReLU", new activations::ActivationReLU()},
+  {"Sigmoid", new activations::ActivationSigmoid()}};
```

Comment:
Seems like this memory will be leaked?

Reply:
Yeah, it makes me uncomfortable, too - even though it is global, so should happen once and stick around for the lifetime of the program. My c++ chops are rusty - not sure what the preferred modern c++ way of handling this is.

I also thought about creating individual global instances and just taking a pointer to them in the map.